cli/src/pcluster/models/compute_fleet_status_manager.py

```diff
@@ -161,18 +131,156 @@ def update_status(self, request_status, in_progress_status, final_status, wait_t
             )
         LOGGER.info("Compute fleet status updated successfully.")
 
-    def _wait_for_status_transition(self, wait_on_status, timeout=300, retry_every_seconds=15):
-        current_status = self.get_status()
-        start_time = time.time()
-        while current_status == wait_on_status and not self._timeout_expired(start_time, timeout):
-            current_status = self.get_status()
-            time.sleep(retry_every_seconds)
+    @abstractmethod
+    def _put_status(self, current_status, next_status):
+        """Set compute fleet status on DB."""
+        pass
 
-        if current_status == wait_on_status:
-            raise TimeoutError("Timeout expired while waiting for status transition.")
+    @abstractmethod
+    def get_status_with_last_updated_time(
+        self, status_fallback=ComputeFleetStatus.UNKNOWN, last_updated_time_fallback=None
+    ):
+        """Get compute fleet status and the last compute fleet status updated time."""
+        pass
 
-        return current_status
+    @staticmethod
+    def get_manager(cluster_name, version, scheduler):
+        """Return compute fleet status manager based on version and plugin."""
+        if packaging.version.parse(version) < packaging.version.parse("3.2.0") and scheduler != "plugin":
```

Comment:
put "3.2.0a0" otherwise if we have an alpha or beta for 3.2.0 this won't work as expected.

Reply:
good catch, tnx!